traversal/selector/exploreRecursive.go

```diff
@@ -192,7 +200,16 @@ func (pc ParseContext) ParseExploreRecursive(n ipld.Node) (Selector, error) {
 	if erc.edgesFound == 0 {
 		return nil, fmt.Errorf("selector spec parse rejected: ExploreRecursive must have at least one ExploreRecursiveEdge")
 	}
-	return ExploreRecursive{selector, selector, limit}, nil
+	var stopCondition *Condition
+	stop, err := n.LookupByString(SelectorKey_StopAt)
+	if err == nil {
+		condition, err := pc.PushParent(erc).ParseCondition(stop)
```

Comment:
This doesn't seem right.  I don't think you want to be doing PushParent twice here, I think you just want `pc.ParseCondition(stop)` and that's it.

Might be good to get a little more testing for this path.  I'd be slightly surprised if this doesn't break something when stopAt is present, as written.

Reply:
removed the pushing of parent context, but i don't think it's used in parsing any of the condition potential union cases so it should be a no-op.